variables.tf

```
@@ -80,3 +80,9 @@ variable "nat_gateway_enabled" {
   description = "Flag to enable/disable NAT gateways for private subnets"
   default     = "true"
 }
+
+variable "auto_assign_public_ip" {
+  type        = "string"
+  default     = "true"
```

```suggestion
  default     = true
```

Let's not use strings for boolean types for TF resources since they expect bool